File: frontend/src/api/payout.ts
import client from './client';

export interface PayoutResponse {
  id: number;
  amount: number;
  wallet_address: string;
  currency?: string;
  card_holder?: string;
  card_number?: string;
  phone_number?: string;
  bank_name?: string;
  status: string;
  created_at: string;
  redirect_url: string;
}

export interface PayoutRequest {
  amount: number;
  wallet_address: string;
  currency?: string;
  card_holder?: string;
  card_number?: string;
  phone_number?: string;
  bank_name?: string;
}

export const requestPayout = (data: PayoutRequest) =>
  client.post<PayoutResponse>('/payout/', data).then((r) => r.data);

export const getPayouts = () =>
  client.get<PayoutResponse[]>('/payout/').then((r) => r.data);

File: backend/schemas/payout.py
from pydantic import BaseModel, field_validator
from decimal import Decimal
from datetime import datetime
from typing import Optional


class PayoutRequest(BaseModel):
    amount: Decimal
    wallet_address: str
    currency: str = "USDT"
    card_holder: Optional[str] = None
    card_number: Optional[str] = None
    phone_number: Optional[str] = None
    bank_name: Optional[str] = None

    @field_validator("amount")
    @classmethod
    def amount_positive(cls, v: Decimal) -> Decimal:
        if v <= 0:
            raise ValueError("Amount must be positive")
        return v


class PayoutResponse(BaseModel):
    id: int
    amount: float
    wallet_address: str
    currency: str = "USDT"
    card_holder: Optional[str] = None
    card_number: Optional[str] = None
    phone_number: Optional[str] = None
    bank_name: Optional[str] = None
    receipt_url: Optional[str] = None
    status: str
    created_at: datetime
    redirect_url: str = ""

    model_config = {"from_attributes": True}
